influxql/parser.go

```diff
@@ -442,7 +444,7 @@ func (p *Parser) parsePrivilege() (Privilege, error) {
 
 // parseSelectStatement parses a select string and returns a Statement AST object.
 // This function assumes the SELECT token has already been consumed.
-func (p *Parser) parseSelectStatement() (*SelectStatement, error) {
+func (p *Parser) parseSelectStatement(tr targetRequirement) (*SelectStatement, error) {
```

Comment:
It's easier to do this validation post-parsing and it's more informative to the user. The caller to `parseSelectStatement()` can return an error saying `"selection target required for continuous query"` instead of `"found FROM, expected INTO`".